### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Anton-Zhytonbaiev**
[DEMO LINK](https://Anton-Zhytonbaiev.github.io/react_clock/)

### CODE REVIEW

**Review by GarryRocksman — APPROVED** [review]
🔥 GJ

File: src/App.tsx [hunk 1]
```diff
@@ -1,39 +1,57 @@
...
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+type State = {
+  hasClock: boolean,
```
Comment (ArturErmolenko) [inline src/App.tsx:12]:
```suggestion
  isClockVisible: boolean,
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,39 +1,57 @@
...
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleLeftClick);
+    document.addEventListener('contextmenu', this.handleRightClick);
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
+
+  componentWillUnmount(): void {
```
Comment (ArturErmolenko) [inline src/App.tsx:32]:
```suggestion
  componentWillUnmount() {
```

**Review by ArturErmolenko — APPROVED** [review]
Well done 👍

**Pull request closed without merging**